---
title: "BIOL7263-03 Project"
author: "Laci Cartmell"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

##### Quick Note: This Repo is used in conjuctiong with a project for BIOL7263-03: Genome Biology


## Background
The goal for the class is to determine if IBD exist for swallow bugs. This project used single read sequences generated using ddRAD-seq on genomic dna from swallow bugs (Cimex vicarius).C. vicarius primarily parasitize cliff swallows which are highly colonial and move frequently between colonies.
My hypothesis is that swallow bugs will not exhibit isolation by distance as the cliff swallows; however, a previous study using mtDNA found that the genetic diversity of swallow bugs from cliff swallow colony sites were negatively associated with distance from ancestral cliffs (Moore, 2016)


## Sequence Generation
Ten swallow bugs were sequenced from 23 and 14 cliff swallow colonies from 2014 and 2021, respectively. The raw sequencing files were generated with a double-digest restriction-associated sequencing (ddRAD-seq) library preparation method based on Peterson et al. (2012). Single-read sequences were generated using Illumina HiSeq 4000. There are 8 raw sequence files that are stored on a google drive folder (email for access)
 

# Post-sequencing Tasks 
- Quality control with FastQC and MultiQC
- Trimming and Demultilexing - process_radtags in Stacks
- Alignment of reads - Integrated method with Stacks
- Post-Alignment 
  - check alignment quality
  - calculate Fst & geographic distance

## Quality Control w/FastQC & MultiQC
Quality checks were done on the raw & demultiplexed files using FastQC. Visualizing hundreds of files can be daunting, which is where MultiQC comes in. It summarizes the statistics from FastQC in an easy to read format. If problem sequences exist, then can be identified and their FastQC files looked at more closely. The results contains summary figures using MultiQC on the browswer. 

Files:

- [fastqc output files](https://github.com/laci-cartmell/cimex.vicarius.ddRAD-seq/tree/main/fastqc)
- [fastqc.sbatch](https://github.com/laci-cartmell/cimex.vicarius.ddRAD-seq/blob/main/scripts/fastqcloop.sbatch)



## Trimming and Demultiplexing
The raw Illumina files need to be demultiplexed so that each individual swallow bug will have its own .fq.gz file. 
The module process_radtags in STACKS demultiplexes the files based on unique barcode sequences. The barcodes are in the required file directory. It is also able to 'clean' reads. The reads must pass a phred quality score of 30 with a sliding window. 

The process_radtags also generates values such as reads and percentage of total reads kept after cleaning. These were looked at in excel and all samples less than 100k samples were discarded. 

Samples were separated into 2014 and 2021 for all future scripts - so most were run twice, once for 2014 & 2021.

Files:

 - [radtag_0301](https://github.com/laci-cartmell/cimex.vicarius.ddRAD-seq/blob/main/scripts/radtags_2203-01.sbatch)
  - [radtag_0302](https://github.com/laci-cartmell/cimex.vicarius.ddRAD-seq/blob/main/scripts/radtags_2203-02.sbatch)
 - [radtag_0308](https://github.com/laci-cartmell/cimex.vicarius.ddRAD-seq/blob/main/scripts/radtags_2203-08.sbatch)
 - [radtag_0901](https://github.com/laci-cartmell/cimex.vicarius.ddRAD-seq/blob/main/scripts/radtags_2209-01.sbatch)   
 - [radtag_0902](https://github.com/laci-cartmell/cimex.vicarius.ddRAD-seq/blob/main/scripts/radtags_2209-02.sbatch)   
 - [radtag_1001](https://github.com/laci-cartmell/cimex.vicarius.ddRAD-seq/blob/main/scripts/radtags_2210-01.sbatch)
 - [radtag_1101](https://github.com/laci-cartmell/cimex.vicarius.ddRAD-seq/blob/main/scripts/radtags_2211-01.sbatch)
 - [radtags_1102](https://github.com/laci-cartmell/cimex.vicarius.ddRAD-seq/blob/main/scripts/radtags_2211-02.sbatch)
 
## Alignment of reads - Integrated method
An integrated alignment is neat becuase it utilizes both de novo and reference alignments. The short-reads are first assembled into loci for each individual, then the loci matched to high quality samples are used to build a catalog that each sample is matched too. Further steps sorts by locus and phase SNPs into genotypes. However, this is the basis of the de novo assembly - the catalog of high quality samples act as the main 'reference' on which to map an individual sequences. 'Stacks' of loci occur where multiple samples contain the same sequence. The number and depth of these stacks are controlled by parameters that are set-up during optimization. 

scripts for de novo: 

  - [ustacks_array_2014](https://github.com/laci-cartmell/cimex.vicarius.ddRAD-seq/blob/main/scripts/ustacks_array_2.sh)
  - [ustacks_array_2021](https://github.com/laci-cartmell/cimex.vicarius.ddRAD-seq/blob/main/scripts/ustacks_array_2021_2.sh)
  - [cstacks](https://github.com/laci-cartmell/cimex.vicarius.ddRAD-seq/blob/main/scripts/cstacks.sbatch)
  - [sstacks](https://github.com/laci-cartmell/cimex.vicarius.ddRAD-seq/blob/main/scripts/sstacks.sbatch)
  - [bwa2bam](https://github.com/laci-cartmell/cimex.vicarius.ddRAD-seq/blob/main/scripts/bwamem.sbatch)
  - [tsv2bam](https://github.com/laci-cartmell/cimex.vicarius.ddRAD-seq/blob/main/scripts/sstacks.sbatch)
  - [gstacks](https://github.com/laci-cartmell/cimex.vicarius.ddRAD-seq/blob/main/scripts/gstacks.sbatch)

The next step for an integrated alignment is to align the consensus sequences to a reference genome. The genome doesn't need to be from the same species, but the more closely related the better. Two closely related genomes were available. I chose to align to C.lectularius as more of my sequences mappped to it than the available C. hempiterus genome. 

I used BWA as my alignment tool to align the loci to the reference genome. Then used samtools to convert to a BAM file. At this time, I also generated summary statistics with flagstat

   - [map_stats_2014](https://github.com/laci-cartmell/cimex.vicarius.ddRAD-seq/blob/main/mappingstats14.txt)
   
   - [map_stats_2021](https://github.com/laci-cartmell/cimex.vicarius.ddRAD-seq/blob/main/mappingstats21.txt)

I used another module from stacks - stacks-integrate-alignments. It takes the coordinates from the BAM file and inserts them into the catalog. 

The final step is to use populations module to generate general statistics and output files for use in R. A genepop file was exported from Stacks.

scripts for integration:

    
  - [integrated-ailgnment](https://github.com/laci-cartmell/cimex.vicarius.ddRAD-seq/blob/main/scripts/integrate.sbatch)
    
  - [populations](https://github.com/laci-cartmell/cimex.vicarius.ddRAD-seq/blob/main/scripts/populations.sbatch)



## From the populations in stacks - it generates a genepop file 

Genepop Files:

  - [2014 genepop](https://github.com/laci-cartmell/cimex.vicarius.ddRAD-seq/blob/main/RequiredFiles/populations.snps14.genepop)
  
  - [2021 genepop](https://github.com/laci-cartmell/cimex.vicarius.ddRAD-seq/blob/main/RequiredFiles/populations.snps21.genepop)


Genepop files are a common file type for population statistics software such as Arelquin. However, I decided to work with Adgenet for generating population statistics as it is also frequenctly used in R but can also store population metadata. The genind were used for general pop stats, but the hierfstat package in R was ultimately used to generate Fst.

The population statistic, Fst, can be used as a measure of genetic distance. It was calculated based on the Wier and Cockerham method.

2014

- [Fst](https://github.com/laci-cartmell/cimex.vicarius.ddRAD-seq/blob/main/Output_R/pairwise.WCfstZ.txt)
- [Fst_UpperLimit](https://github.com/laci-cartmell/cimex.vicarius.ddRAD-seq/blob/main/Output_R/boot.fst.ul.txt)
- [Fst_LowerLimit](https://github.com/laci-cartmell/cimex.vicarius.ddRAD-seq/blob/main/Output_R/boot.fst.ll.txt)

2021

- [Fst](https://github.com/laci-cartmell/cimex.vicarius.ddRAD-seq/blob/main/Output_R/pairwise.WCfstZ.21.txt)
- [Fst_UpperLimit](https://github.com/laci-cartmell/cimex.vicarius.ddRAD-seq/blob/main/Output_R/boot.fst.ul21.txt)
- [Fst_LowerLimit](https://github.com/laci-cartmell/cimex.vicarius.ddRAD-seq/blob/main/Output_R/boot.fst.ll21.txt)



The geographic distance was calculated using coordinates for each colony location. The distance between each colony was calculated using Geosphere (distance based on WGS1984 geodesic).Then a pairwise distance matrix was calculated.

  [geo_distance R script](https://github.com/laci-cartmell/cimex.vicarius.ddRAD-seq/blob/main/Output_R/boot.fst.ll21.txt)
  
  [fst_calculation R script](https://github.com/laci-cartmell/cimex.vicarius.ddRAD-seq/blob/main/scripts/FST_calculation.R)


## The Results

The pairwise Fst were low, and most were not significantly different from other colones. 

![FST_2014](https://github.com/laci-cartmell/cimex.vicarius.ddRAD-seq/blob/main/Output_R/pairwisefst_2014.pdf)
![FST_2021](https://github.com/laci-cartmell/cimex.vicarius.ddRAD-seq/blob/main/Output_R/pfst2_21.pdf)

The mantel tests were non-significant for 2014 and 2021. The Mantel r-value was 0.0656 for 2014 and -0.0159 for 2021. 

![IBD_2014](https://github.com/laci-cartmell/cimex.vicarius.ddRAD-seq/blob/main/Output_R/ibd_2014.svg)
![IBD_2021](https://github.com/laci-cartmell/cimex.vicarius.ddRAD-seq/blob/main/Output_R/IBD_2021.png)


Script: 
[mantel_test R script](https://github.com/laci-cartmell/cimex.vicarius.ddRAD-seq/blob/main/scripts/manteltest.R)

## Conclusion

Swallow bugs are not more genetically different with geographic distance between colonies.

